src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```diff
@@ -2,32 +2,21 @@
 
   <PropertyGroup Condition="'$(RuntimeIdentifier)' != ''">
     <!-- Define the name of the runtime specific compiler package to import -->
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('win'))">win</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('osx'))">osx</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('maccatalyst'))">maccatalyst</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('iossimulator'))">iossimulator</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('ios'))">ios</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('tvossimulator'))">tvossimulator</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('tvos'))">tvos</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('freebsd'))">freebsd</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == '' and ($(RuntimeIdentifier.StartsWith('linux-musl')) or $(RuntimeIdentifier.StartsWith('alpine')))">linux-musl</OSIdentifier>
-    <OSIdentifier Condition="'$(OSIdentifier)' == ''">linux</OSIdentifier>
```

Comment:
This took care of all the specific Linux RIDs (Ubuntu, etc.). Are those RIDs going to be broken now? (It is not necessary a bad thing.)

Reply:
Good point. We can use `NETCoreSdkPortableRuntimeIdentifier` which provides linux-x64 with non-portable RIDs like ubuntu.22.04-x64 etc. This is what SDK and installer are using for runtime packs that are only available with portable flavor:
https://github.com/dotnet/runtime/blob/d702914b944a7ad20fd9119af4e855863bbedc08/src/coreclr/.nuget/Microsoft.NET.Sdk.IL/targets/Microsoft.NET.Sdk.IL.targets#L30